### PR TITLE
Adjust loading of default openssl certs

### PR DESCRIPTION
**Pull request by davisagli**
Based on @tiran's recommendations in #32 

This does not yet eagerly load certs from capath on Mac and Windows so they can be passed to the OS APIs, which is also discussed there but somewhat orthogonal.

### CODE REVIEW

**Review by sethmlarson — COMMENTED** [review]
Thanks for this! Two comments:

File: src/truststore/_openssl.py [hunk 1]
```diff
@@ -1,49 +1,65 @@
...
 
     ctx.verify_mode = ssl.CERT_REQUIRED
     ctx.check_hostname = True
 
 
+def _capath_contains_certs(capath: str) -> bool:
+    """Check whether capath exists and contains certs in the expected format."""
+    if not os.path.isdir(capath):
+        return False
+    for name in os.listdir(capath):
```
[the remark this answers is not in the document]
Comment (davisagli) [reply src/truststore/_openssl.py:52]: @sethmlarson Thanks for the suggestion, but I'm not sure I understand the threat model you're trying to protect against. (tell me more via dm if you prefer?) I benchmarked and there's still a pretty big difference in timing between a nonexistent directory and an empty one when catching the OSError:

```python
import errno
import os
import timeit


bogus_path = os.path.expanduser("~/bogus")
empty_path = os.path.expanduser("~/empty")
if not os.path.exists(empty_path):
    os.mkdir(empty_path)

def check_with_isdir(d):
    if not os.path.isdir(d):
        return False
    for path in os.listdir(d):
        return True
    return False

print(timeit.timeit(lambda: check_with_isdir(bogus_path), number=10000))
# 0.032685505
print(timeit.timeit(lambda: check_with_isdir(empty_path), number=10000))
# 0.252140689

def check_with_exc(d):
    try:
        for path in os.listdir(d):
            return True
    except OSError as e:
        if e.errno == errno.ENOENT:
            return False
        raise
    return False

print(timeit.timeit(lambda: check_with_exc(bogus_path), number=10000))
# 0.03667674899999995
print(timeit.timeit(lambda: check_with_exc(empty_path), number=10000))
# 0.18221601999999998
```

[the remark this answers is not in the document]
Comment (sethmlarson) [reply src/truststore/_openssl.py:52]: You're right, there's no potential for filesystem timing silliness here.

File: src/truststore/_openssl.py [hunk 2]
```diff
@@ -1,49 +1,65 @@
...
+    if defaults.cafile or (defaults.capath and _capath_contains_certs(defaults.capath)):
+        ctx.set_default_verify_paths()
     else:
-        for cadir in _CA_DIRS:
-            if os.path.isdir(cadir):
-                ctx.load_verify_locations(capath=cadir)
+        # cafile from OpenSSL doesn't exist
+        # and capath from OpenSSL doesn't contain certs.
+        # Let's search other common locations instead.
+        cafile = None
```
Comment (sethmlarson) [inline src/truststore/_openssl.py:38]: nit: Do we need to initialize `cafile` here since it's not used below this block?

Comment (davisagli) [reply src/truststore/_openssl.py]: This was vestigial; fixed

**Review by sethmlarson — APPROVED** [review]
LGTM, thanks! :shipit:

**Pull request merged**